Adapt array shapes for psalm5

mstilkerich opened the pull request:
Array shapes in psalm 5 do not allow extra keys anymore, unless explicitly specified in the shape definition using `...`.
See 'sealed vs unsealed array shapes' in https://psalm.dev/articles/psalm-5

This means that psalm 5 will throw errors for example when a Guzzle client is constructed with additional request options.

This PR adapts the current array shape definitions in the stubs to work with psalm 5.

This breaks backwards compat with earlier psalm versions which do not support the '...' extension in array shapes.

mstilkerich committed:
Adapt array shapes for psalm5

This breaks backwards compat with earlier psalm versions which do not
support the '...' extension in array shapes.

alexeyshockov commented on composer.json:
```diff
@@ -10,7 +10,7 @@
     ],
     "require": {
-        "vimeo/psalm": "^3 || ^4 || ^5"
+        "vimeo/psalm": "^5"
```
That means a breaking change, so I will bump the major version number. Probably time to release 1.x.

alexeyshockov commented on stubs/Handler/CurlMultiHandler.php:
```diff
@@ -9,7 +9,7 @@
 {
     /**
-     * @param array{handle_factory?:CurlFactoryInterface,select_timeout?:int|float} $options
+     * @param array{handle_factory?:CurlFactoryInterface,select_timeout?:int|float,options?:array} $options
```
```suggestion
     * @param array{handle_factory?:CurlFactoryInterface, select_timeout?:int|float, options?:array} $options
```

Could you also format all the `array{}` types to include a space after a colon? Like this one, to maintain code code style for types, ha-ha.

alexeyshockov reviewed (approved):
Great, would just appreciate a code style change!

mstilkerich committed:
Reformat annotations

- Add spaces after commas separating array shape elements
- Add spaces after colons in array shape elements
- Add = character to psalm-type annotations

mstilkerich commented:
Hello, I edited the annotations as suggested. I also noticed missing `=` in many `@psalm-type` annotations, which appears to be accepted by psalm, but according to the docs an `=` character is expected in that annotation.

https://psalm.dev/docs/annotating_code/supported_annotations/#psalm-type

mstilkerich commented:
Would you mind publishing an updated release?

alexeyshockov commented:
@mstilkerich, thanks a lot for your changes!

mstilkerich commented:
Well thanks for your work on providing this plugin :-)

Great, now my CI workflow is working again!
